src/Util/CliOutputHelper.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+/**
+ * Tools used to enhance maker command output.
+ *
+ * For additional context with Symfony CLI EnvVars, see
+ * https://github.com/symfony-cli/symfony-cli/pull/231
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+final class CliOutputHelper
```

Comment:
Could this be integrated directly into symfony/console ?

Reply:
I don't see why not. We would need to add getters for the version & binary name (if not now, in the future) should the need arise where we need to conditionally change messaging in maker for some reason..

I can get a PR going in `symfony/console` to get this in... Should it be done as helper like we have here or integrated directly into another part of an existing class?

Reply:
Side note - do we have time to get it in for 6.2?

Reply:
I would not put this directly in symfony/console. We currently have nothing exposing just this info in the component and `php bin/console` is still specific to the Flex skeleton.

However, `symfony/console` has an internal logic that would benefit from checking `SYMFONY_CLI_BINARY_NAME` when replacing the placeholders in help texts.

Reply:
This is exactly the patch I had in mind. But ideally, we would have an env variable covering the whole symfony-cli command instead of hard-coding `console` in symfony/console, as this would cause issues with `symfony composer` for instance.
and I'm wondering of the effect when running `symfony php` as the script itself is not wrapped there. so this requires more discussion before being readme (and it won't be for 4.4 as it reached EOM)

Reply:
exactly the uses case I had in mind and lead me to not submit the patch **yet** :D 
yeah I only targeted 4.4 by lazyness ;)

Reply:
I would say that `symfony-cli` should expose 2 env variables to solve that:

- an env variable exposing the wrapper command (to be decided whether we expose `symfony console` or just `console` and expect the consumer to combine it with the existing env var for the binary name)
- an env variable exposing whether the wrapper command is only wrapping PHP itself or also the PHP script (to decide whether it replaces the script name or only prefix it in symfony/console)